.gitlab-ci.yml

```diff
@@ -24,83 +24,87 @@ stages:
-    - mkdir -p bus/$CI_BUILD_NAME/cran/src/contrib
+    - mkdir -p bus/$CI_JOB_NAME/cran/src/contrib
     - Rscript -e 'mirror.packages(dcf.dependencies("DESCRIPTION", "all"), repos=Sys.getenv("CRAN_MIRROR"), repodir="bus/mirror-packages/cran")'
-    - rm bus/$CI_BUILD_NAME/cran/src/contrib/PACKAGES.rds ## fallback to PACKAGES dcf so available.packages:3.4.4 works
     - Rscript -e 'sapply(simplify=FALSE, setNames(nm=Sys.getenv(c("R_REL_VERSION","R_DEVEL_VERSION","R_OLDREL_VERSION"))), function(binary.ver) mirror.packages(type="win.binary", dcf.dependencies("DESCRIPTION", "all"), repos=Sys.getenv("CRAN_MIRROR"), repodir="bus/mirror-packages/cran", binary.ver=binary.ver))'
   <<: *artifacts
 
-# mirror-other-packages: ## mirror integration suggests from pkgs at the top of inst/tests/other.Rraw;  off now #5274
-#  stage: dependencies
-#  tags:
-#    - linux
-#  image: registry.gitlab.com/jangorecki/dockerfiles/r-base-dev
-#  cache:
-#    paths:
-#      - bus/$CI_BUILD_NAME/cran
-#  script:
-#    - echo 'source(".ci/ci.R")' >> .Rprofile
-#    - mkdir -p bus/$CI_BUILD_NAME/cran/src/contrib
-#    - Rscript -e 'eval(parse("inst/tests/other.Rraw", n=1L)); mirror.packages(pkgs, repos=Sys.getenv("CRAN_MIRROR"), repodir="bus/mirror-other-packages/cran")'
-#  <<: *artifacts
-
-build: ## build data.table sources as tar.gz archive
+## build
+# sources as tar.gz archive
+# build vignettes
+build:
   stage: build
   tags:
     - linux
-  image: registry.gitlab.com/jangorecki/dockerfiles/r-builder
+  image: registry.gitlab.com/jangorecki/dockerfiles/r-base ## r-base-gcc after rstudio/markdown#108
   needs: ["mirror-packages"]
   before_script:
     - Rscript -e 'install.packages(c("knitr","rmarkdown"), repos=file.path("file:",normalizePath("bus/mirror-packages/cran")), quiet=TRUE)'
     - rm -r bus
     - sed -i '/^[[:space:]]*$/d' ./DESCRIPTION ## make last line end abruptly; i.e. without a final \n
-    - echo "Revision:" $CI_BUILD_REF >> ./DESCRIPTION
+    - echo "Revision:" $CI_COMMIT_SHA >> ./DESCRIPTION
   script:
     - R CMD build .
-    - mkdir -p bus/$CI_BUILD_NAME/cran/src/contrib
-    - mv $(ls -1t data.table_*.tar.gz | head -n 1) bus/$CI_BUILD_NAME/cran/src/contrib/.
+    - mkdir -p bus/$CI_JOB_NAME/cran/src/contrib
+    - mv $(ls -1t data.table_*.tar.gz | head -n 1) bus/$CI_JOB_NAME/cran/src/contrib/.
     - Rscript -e 'tools::write_PACKAGES(contrib.url("bus/build/cran"), fields="Revision", addFiles=TRUE)'
-    - rm bus/$CI_BUILD_NAME/cran/src/contrib/PACKAGES.rds ## fallback to PACKAGES dcf so available.packages:3.4.4 works
   <<: *artifacts
 
+## install deps aliases
 .test-install-deps: &install-deps
-  - Rscript -e 'source(".ci/ci.R"); install.packages(dcf.dependencies("DESCRIPTION", which="most"), quiet=TRUE)'
+  - Rscript -e 'source(".ci/ci.R"); install.packages(dcf.dependencies("DESCRIPTION", which="all"), repos=file.path("file:", normalizePath("bus/mirror-packages/cran", mustWork=FALSE)), quiet=TRUE)'
```

Comment:
I would start (e.g.) `.ci/scripts/` and add `.ci/scripts/test-install-deps.R` with:

```r
source(".ci/ci.R")
install.packages(
  dcf.dependencies("DESCRIPTION", which="all"),
  repos=file.path("file:", normalizePath("bus/mirror-packages/cran", mustWork=FALSE)),
  quiet=TRUE
)
```

This will be more readable and avoid duplication (and hence double maintenance/risk of getting out of sync) across `test-intall-deps` and `test-install-deps-win` jobs.

Reply:
Or, we can find a way to put `RUN_SCRIPT=if(win) Rscript.exe else Rscript` into env and unify the jobs.

Reply:
Quoting arguments from shell also differs

Reply:
> I would start (e.g.) `.ci/scripts/` and add `.ci/scripts/test-install-deps.R` with:
> 
> ```r
> source(".ci/ci.R")
> install.packages(
>   dcf.dependencies("DESCRIPTION", which="all"),
>   repos=file.path("file:", normalizePath("bus/mirror-packages/cran", mustWork=FALSE)),
>   quiet=TRUE
> )
> ```
> 
> This will be more readable and avoid duplication (and hence double maintenance/risk of getting out of sync) across `test-intall-deps` and `test-install-deps-win` jobs.

Unless we will need distinction for some reason, then we ended up having two scripts :)

Reply:
> Quoting arguments from shell also differs

Can't we unify if there's no env inside `"`? Anyway, if the shell command is just `Rscript -e "source(...)"` it will still be easier to glance & see it looks ~OK.

> Unless we will need distinction for some reason, then we ended up having two scripts :)

Looks like not the case yet, we can do some `.Platform`/`Sys.info()` switching to handle the small differences. Still almost all of the logic between the two jobs is the same, I find it more confusing needing to read long one-line scripts without syntax highlighting.